### PR TITLE
The Pururin.io extension.

### LINKED ISSUE #92: [Multiple] Pururin.us

### PR DESCRIPTION
**Pull request by BlackNuziGuy**
Closes #92 
I obtained the icons from the repo of @as280093.

### CODE REVIEW

File: src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+class PuruRin : ParsedHttpSource() {
+
+    private fun pagedRequest(url: String, page: Int, queryString: String? = null): Request {
+        // The site redirects page 1 -> url-without-page so we do this redirect early for optimization
+        val builtUrl = if (page == 1) url else "${url}browse/newest?page=$page"
+        return GET(if (queryString != null) "$url$queryString" else builtUrl)
+    }
+
+    override val name = "PuruRin"
```
Comment (arkon) [inline src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt]: Nitpick: just "Pururin" to match the site. Same for the class name.

File: src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    override fun popularMangaSelector() = latestUpdatesSelector()
+
+    override fun popularMangaNextPageSelector() = latestUpdatesNextPageSelector()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        var url: String? = null
+        var queryString: String? = null
+        fun requireNoUrl() = require(url == null && queryString == null) {
+            "[ROBLOX OOF SOUND]"
```
Comment (arkon) [inline src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt]: Wat

File: src/en/pururin/build.gradle [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+ext {
+    appName = 'Tachiyomi: Pururin'
+    pkgNameSuffix = 'en.pururin'
+    extClass = '.PuruRin'
+    extVersionCode = 2
```
Comment (arkon) [inline src/en/pururin/build.gradle]: Why 2?

File: src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt (hunk 1 above)
Comment (BlackNuziGuy) [reply src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt]: Done

File: src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt (hunk 2 above)
Comment (BlackNuziGuy) [reply src/en/pururin/src/eu/kanade/tachiyomi/extension/en/pururin/PuruRin.kt]: Oops. Forgot to remove that

File: src/en/pururin/build.gradle (hunk 3 above)
Comment (BlackNuziGuy) [reply src/en/pururin/build.gradle]: The first one did not work properly.
So I made the second one;
I'll just change it back to 1.

**Pull request merged**